### PR TITLE
Enable temperature profile

### PR DESCRIPTION
**Pull request by aditjind**
*Fixes #:* Enable Node Temperature RCA. This PR enables the Node Temperature RCA on the cluster. 

*Description of changes:* This commit enables the temperature Profile and refactors the code which removes the healthy/unhealthy context of the dimensional RCAs. Also adds comments and checks for values of Flow Units in the Individual RCAs & Group RCAs. Also change the use to ShardIndexKey instead of Shard ID and Index ID separately. 

*Tests:* Tested on Docker. Sample Output: 

**Node 1:** 

```
[root@391ae37d34b6 tmp]# curl --url "localhost:9600/_opendistro/_performanceanalyzer/rca?name=NodeTemperatureRca&local=true" -XGET | python -m json.tool 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   459  100   459    0     0   2244      0 --:--:-- --:--:-- --:--:--  2250
{
    "NodeTemperatureRca": [
        {
            "CompactNodeSummary": [
                {
                    "CPU_Utilization_mean": 0,
                    "CPU_Utilization_num_shards": 0,
                    "CPU_Utilization_total": 0.0,
                    "Heap_AllocRate_mean": 0,
                    "Heap_AllocRate_num_shards": 0,
                    "Heap_AllocRate_total": 0.0,
                    "Shard_Size_In_Bytes_mean": 3,
                    "Shard_Size_In_Bytes_num_shards": 3,
                    "Shard_Size_In_Bytes_total": 23449682.0,
                    "host_address": "172.29.0.2",
                    "node_id": "SqlWFl9BR02Bz0gVcFS8qQ"
                }
            ],
            "rca_name": "NodeTemperatureRca",
            "state": "unknown",
            "timestamp": 1605582526245
        }
    ]
}

```

Node 2: 

```
[root@ef8562eeb2ed tmp]# curl --url "localhost:9600/_opendistro/_performanceanalyzer/rca?name=NodeTemperatureRca&local=true" -XGET | python -m json.tool 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   459  100   459    0     0   9945      0 --:--:-- --:--:-- --:--:-- 10200
{
    "NodeTemperatureRca": [
        {
            "CompactNodeSummary": [
                {
                    "CPU_Utilization_mean": 0,
                    "CPU_Utilization_num_shards": 0,
                    "CPU_Utilization_total": 0.0,
                    "Heap_AllocRate_mean": 0,
                    "Heap_AllocRate_num_shards": 0,
                    "Heap_AllocRate_total": 0.0,
                    "Shard_Size_In_Bytes_mean": 5,
                    "Shard_Size_In_Bytes_num_shards": 2,
                    "Shard_Size_In_Bytes_total": 15918349.0,
                    "host_address": "172.29.0.3",
                    "node_id": "XyinFukETiaOdwfSkGnM8w"
                }
            ],
            "rca_name": "NodeTemperatureRca",
            "state": "unknown",
            "timestamp": 1605582517354
        }
    ]
}
```

`AllTemperatureDimensions` have also been enabled. 

*If new tests are added, how long do the new ones take to complete*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by yojs** [comment]
can we make sure that the temperature profile is working with the new perftop dashboard we added for it ?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/temperature/NodeLevelDimensionalSummary.java [hunk 1]
```diff
@@ -60,6 +62,14 @@
     private final NodeLevelZoneSummary[] zoneProfiles;
     private int numberOfShards;
 
+    // Mean Temperature is a mean of the normalized heat of the resource used across shards on the node.
+    // e.g. if there are 10 shards on the node and the normalized sum of the resource used
```
Comment (yojs) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/temperature/NodeLevelDimensionalSummary.java:66]: Awesome ! great comment

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/temperature/NodeLevelDimensionalSummary.java [hunk 2]
```diff
@@ -60,6 +62,14 @@
...
+    // across shards is 33. The mean would be 3.3
+    // This aim is the balance this parameter across the nodes and have as little delta
```
Comment (yojs) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/temperature/NodeLevelDimensionalSummary.java:68]: typo: this -> the ?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/core/temperature/ShardStore.java [hunk 3]
```diff
@@ -32,44 +31,26 @@
...
-     * The key for the outer map is indexName. The key for inner map is the ShardID. Given an
-     * indexName and shardId, a shard can be uniquely identified.
-     *
-     * <p>TODO: Try replace with 'IndexShardKey' after
-     * https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/pull/141
-     * is merged.
+     * The key for the map is indexShardKey Given an IndexShardKey a shard can be uniquely identified.
      */
-    Map<String, Map<Integer, ShardProfileSummary>> list;
+    Map<IndexShardKey, ShardProfileSummary> shardToShardProfileMap;
```
Comment (yojs) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/core/temperature/ShardStore.java:36]: This is a good change. Thanks a lot

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java [hunk 4]
```diff
@@ -915,13 +915,11 @@ private synchronized JsonElement readTemperatureProfileRca(String rca) {
...
-            Integer.class);
-        SelectJoinStep<Record> query = SQLiteQueryUtils
-            .buildSummaryQuery(create, ClusterTemperatureSummary.TABLE_NAME,
-                mostRecentRecord.get(primaryKeyField),
-                foreignKeyField);
+        SelectJoinStep<Record> query = SQLiteQueryUtils.buildSummaryQuery(
+            create,
+            ClusterTemperatureSummary.TABLE_NAME,
+            mostRecentRecord.get(primaryKeyField),
+            primaryKeyField);
```
Comment (yojs) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java:922]: Let's use the new Object persisting mechanism

**Review by yojs — APPROVED** [review]

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java (hunk 4 above)
Comment (khushbr) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java:921]: @yojs  : Can you add more details here? Let's create a separate Issue and work on it given this PR is old. @yujias0706

**Review by khushbr — APPROVED** [review]

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java (hunk 4 above)
Comment (yujias0706) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java:921]: Created issue: https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/issues/559

**Review by yojs — APPROVED** [review]

**Pull request merged**